Update JumpProcesses documentation (#372)
SciML/JumpProcesses.jl: docs/src/tutorials/simple_poisson_process.md

may occur, can skip directly to the [second tutorial](@ref ssa_tutorial) for a
tutorial covering similar material but focused on the SIR model.

!!! note
    
    If you have a background in point process theory, we recommend that
    you read our example on [modelling temporal point processes with
    JumpProcesses](@ref tpp_tutorial)

> 

Reviewer: I would just merge this into the first paragraph for parallel construction and say something as the new last sentence like "Readers with a background in point process theory can skip directly to [your new tutorial]."

Author: Fixed.

```

We next create a [`JumpProblem`](@ref) that wraps the discrete problem, and

> specifies which sampling algorithm, called an aggregator in JumpProcesses, to
> use for determining next jump times (and in the case of multiple possible jumps
> the next jump type). We call them aggregators because these algorithms are
> methods for aggregating a set of jumps to determine the next jump time.  Here
> we use the classical `Direct` method, proposed by Gillespie in the chemical
> reaction context, but going back to earlier work by Doob and others (and also
> known as Kinetic Monte Carlo in the physics literature)

Reviewer: ```suggestion
specifies which sampling algorithm, called an aggregator in JumpProcesses, to
use for determining next jump times (and in the case of multiple possible jumps
the next jump type). We call them aggregators because these algorithms are
methods for aggregating a set of jumps to determine the next jump time and type.  Here
we use the classical `Direct` method, proposed by Gillespie in the chemical
reaction context, but going back to earlier work by Doob and others (and also
known as Kinetic Monte Carlo in the physics literature)
```

```

As the death rate is constant *between* jumps, we can encode this process as a

> second `ConstantRateJump`. Overall, we keep track of four different processes,
> the population size ``N``, total deaths ``D``, births ``Y_b`` and deaths
> ``Y_d``. ``N`` and ``D`` are represented by ``u`` in our code, ``Y_b`` by `crj`
> and ``Y_d`` by `death_crj`. We then construct the corresponding problems,
> passing both jumps to `JumpProblem`, and can solve as before

Reviewer: ```suggestion
second `ConstantRateJump`. Overall, we keep track of four different processes,
the population size ``N(t)``, total deaths ``D(t)``, births ``Y_b(t)`` and deaths
``Y_d(t)``. The vector ``(N(t), D(t))`` is represented by ``u`` in our code, with ``Y_b(t)`` encoded by `crj`
and ``Y_d(t)`` by `death_crj`. We then construct the corresponding problems,
passing both jumps to `JumpProblem`, and can solve as before
```

vrj1 = VariableRateJump(rate1, affect1!; lrate, urate, rateinterval)
```

Finally, to efficiently simulate the new jump process, we must also
specify a dependency graph. This indicates when a given jump occurs, which
jumps in the system need to have their rates and/or rate bounds
recalculated (for example, due to depending on changed components in `u`).
Internally, JumpProcesses preserves the relative ordering of jumps of each
distinct type, but always reorders all `ConstantRateJump`s to appear

> before any `VariableRateJump`s. Irrespective of how `JumpProblem` is
> initialized, we can think that internally the jumps will get ordered as
> vector `[deathcrj, vrj1]`. This vector of jumps is not the same as our
> state variable `u`.

Reviewer: ```suggestion
before any `VariableRateJump`s. Irrespective of how `JumpProblem` is
initialized, internally the jumps will be ordered as the vector `[deathcrj, vrj1]`. Note, this vector of the jumps is distinct from our state variable vector, `u`.
```
